photutils/psf/daogroup.py

```diff
@@ -0,0 +1,129 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+"""Functions for implementing the DAOGROUP algorithm proposed by Stetson 1987.
+"""
+
+from __future__ import division
+import numpy as np
+from astropy.table import Column, Table, vstack
+
+__all__ = ['daogroup']
+
+def daogroup(starlist, crit_separation):
+    """
+    This is an implementation of the DAOGROUP algorithm presented by
+    Stetson (1987).
+
+    DAOGROUP divides an entire starlist into sets of distinct, self-contained
+    groups of mutually overlapping stars.
+
+    GROUP accepts as input a list of stars and determines which stars are
+    close enough to be capable of adversely influencing each others' profile
+    fits.
+
+    Parameters
+    ----------
+    starlist : `~astropy.table.Table` or array-like
+        List of stars positions.
```

Comment:
I know we've talked about the API quite a bit. I wonder if time has come to *require* and astropy.table as input. They are not hard to make if you have something that's array like already. Most users will use this as part of the pipeline anyway and the other functions in photutils return astropy.table.Table objects.
In my mind, the "array like" is a left-over from the beginning of photutils when the design was not quite there for the Table structure. But now, that's all done.
I bet that photutils total line count can be decreased by 5% if we just use Table everywhere, because if  will save us a lot of if .. else.. . Also, it's safer and easier to set up (was the order in the array x,y, flux or was it flux, x, y ?).

I understand that existing photutils methods want to be backwards compatible, but once we write something new, we could decide to allow and and only one type on input in line with the "Zen of Python": 
```
>>> import this
The Zen of Python, by Tim Peters

Beautiful is better than ugly.
Explicit is better than implicit.
Simple is better than complex.
Complex is better than complicated.
Flat is better than nested.
Sparse is better than dense.
Readability counts.
Special cases aren't special enough to break the rules.
Although practicality beats purity.
Errors should never pass silently.
Unless explicitly silenced.
In the face of ambiguity, refuse the temptation to guess.
There should be one-- and preferably only one --obvious way to do it.
Although that way may not be obvious at first unless you're Dutch.
Now is better than never.
Although never is often better than *right* now.
If the implementation is hard to explain, it's a bad idea.
If the implementation is easy to explain, it may be a good idea.
Namespaces are one honking great idea -- let's do more of those!
```
Tables are clearly more explicit and they there would be one and only one way.